### ISSUE TITLE
The heap implementation does not properly sort items

### ISSUE THREAD

**Issue by eliaslevy** [emirpasic/gods#74]
E.g.
```go
	heap := binaryheap.NewWithIntComparator()
	heap.Push(5)
	heap.Push(2)
	heap.Push(3)
	heap.Push(1)
	heap.Push(4)
	fmt.Println(heap.Values())
```
Results in `[1 2 3 5 4]`

**Comment by eliaslevy** [comment]
Reading the tests it seems `Values()` and the iterator are not expected to be ordered.  I got the opposite impression from the documentation.  If that is the case, then there is bug.  But it would be a lot more useful of the iterator returned items in the sort order.

**Comment by emirpasic** [comment]
@eliaslevy 

You are right that the values are not expected to be ordered in the way you expected it and that's exactly how a binary heap works: https://www.cs.cmu.edu/~adamchik/15-121/lectures/Binary%20Heaps/heaps.html

> A binary heap is a complete binary tree which satisfies the heap ordering property. The ordering can be one of two types:

> the min-heap property: the value of each node is greater than or equal to the value of its parent, with the minimum-value element at the root.
the max-heap property: the value of each node is less than or equal to the value of its parent, with the maximum-value element at the root.

In case you need to sort them, you can sort them manually using: https://github.com/emirpasic/gods#sort

Otherwise, if you'd like the data structure to keep the elements sorted, then you probably would like to use some other type of data structure, e.g. TreeSet https://github.com/emirpasic/gods#treeset

Cheers
